src/util/event_emitter.js

```diff
@@ -0,0 +1,140 @@
+export class EventEmitter {
+  constructor() {
+    this._handlers = [];
+  }
+
+  /**
+   * Adds the event name and handler function to the end of the handlers array.
+   * No checks are made to see if the handler has already been added.
+   * Multiple calls passing the same combination of event name and handler will result in the handler being added,
+   * and called, multiple times.
+   * @param {String} event
+   * @param {Function} handler
+   * @returns {EventEmitter}
+   */
+  on(event, handler) {
+    this._handlers.push({
+      event,
+      handler
+    });
+
+    return this;
+  }
+
+  _onceWrapper(...args) {
+    if (!this.fired) {
+      this.target.off(this.event, this.wrapFn);
+      this.fired = true;
+      Function.prototype.apply.call(this.handler, this.target, [...args]);
+    }
+  }
+
+  _onceWrap(event, handler) {
+    const state = {
+      fired: false,
+      wrapFn: undefined,
+      target: this,
+      event,
+      handler
+    };
+    const wrapped = this._onceWrapper.bind(state);
```

Comment:
```suggestion
  const onceWrapper = function(...args) {
    if (!this.fired) {
      this.target.off(this.event, this.wrapFn);
      this.fired = true;
      Function.prototype.apply.call(this.handler, this.target, [...args]);
    }
  }
  const wrapped = onceWrapper.bind(state);
```

If use of `this` can be confusing inside `onceWrapper`, It can be place outside `EventEmitter` class as helper function too